rfd/0010-api.md

```diff
@@ -35,89 +34,211 @@ func connectClient() (*auth.Client, error) {
+ - Unstable - The API can start at `v0`, which implies instability. When we are ready to guarantee the API's backwards compatibility and stability, it can be upgraded to `v1`.
+   - If we choose to do this, I would suggest that we upgrade to `v1` once the old client is completely deprecated and the new client is used internally.
+ - Flexibility - The API will only be released when it is necessary to do so. This can be more or less frequent than Teleport, depending on what features are needed and when.
+ - Backwards Compatibility - Since the API will follow semver, it will continue to be backwards compatible with each minor release, even if we release several major versions of Teleport between those minor releases.
+
+Cons:
+ - Compatibility - It might not be immediately obvious based on the client version what server/cli versions are compatible.
+   - A [compatibility matrix](https://github.com/kubernetes/client-go#compatibility-matrix) could be added to the docs and `api/client` to help reason about compatibility. 
+   - One solution to this problem is to version the api as `v1.[teleport-major-version].[api-minor/patch-version]`. With this strategy, any `v1.M.X` Api version will work with any `vM.Y.Z` Teleport version, while still getting the benefits of having a separately managed version.
+   - Since the client will have automatic server version checks, users should be able to quickly catch incorrect version usage regardless.
+ - Release management - The API will need to be released on its own schedule with its own versions and branches. This will burden release managers with basically a new product to worry about. 
+   - In the long run, this may be less of a burden than it might seem at first. With `Release option 1`, we may have 20 or so new versions of `teleport/api` released, while with `Release option 2`, most likely there will only be a couple of new versions.
+
+This option provides the most flexibility in API releases. It also gives us the option to start with `v0`, with no stability/compatibility promises, in order to try out a strategy before committing fully to it. If treating the API as a separately versioned product is too burdensome, we can immediately transition to another release strategy.
+
+##### Release option choice
+
+In my opinion, `Release option 1` is the better choice. It will be much easier to get started with its versioning since it can follow all the same release processes in place. `Release option 2` brings many more complexities into the discussion, such as new compatibility rules for interfacing with the API and new branches and tags for the api. 
+
+However, it may be that releasing the API and other futures modules separately (Option 2) will give less headaches at scale, but this is hard for me to measure. For that reason, it would make more sense for the release managers to choose a strategy. So far the consensus is to go with Release Option 1.
```

Comment:
```suggestion
The consensus is `Release option 1`.

It will be much easier to get started with its versioning since it can follow all the same release processes in place. `Release option 2` brings many more complexities into the discussion, such as new compatibility rules for interfacing with the API and new branches and tags for the api. 
```